zerver/lib/actions.py

```diff
@@ -5409,31 +5409,29 @@ def send_change_stream_message_retention_days_notification(
 
     with override_language(stream.realm.default_language):
         if old_value == Stream.MESSAGE_RETENTION_SPECIAL_VALUES_MAP["unlimited"]:
-            notification_string = _(
-                "{user} has changed the [message retention period](/help/message-retention-policy) for this stream:\n"
-                "* **Old retention period**: Forever\n"
-                "* **New retention period**: {new_value} days\n"
-                "Messages will be automatically deleted after {new_value} days."
-            )
-            notification_string = notification_string.format(user=user_mention, new_value=new_value)
+            old_retention_period = "Forever"
+            new_retention_period = f"{new_value} days"
+            summary_line = f"Messages will be automatically deleted after {new_value} days."
         elif new_value == Stream.MESSAGE_RETENTION_SPECIAL_VALUES_MAP["unlimited"]:
-            notification_string = _(
-                "{user} has changed the [message retention period](/help/message-retention-policy) for this stream:\n"
-                "* **Old retention period**: {old_value} days\n"
-                "* **New retention period**: Forever\n"
-                "Messages will be kept **Forever** and will **NOT** get automatically deleted after {old_value} days."
-            )
-            notification_string = notification_string.format(user=user_mention, old_value=old_value)
+            old_retention_period = f"{old_value} days"
+            new_retention_period = "Forever"
+            summary_line = f"Messages will be kept **Forever** and will **NOT** get automatically deleted after {old_value} days."
         else:
-            notification_string = _(
-                "{user} has changed the [message retention period](/help/message-retention-policy) for this stream:\n"
-                "* **Old retention period**: {old_value} days\n"
-                "* **New retention period**: {new_value} days\n"
-                "Messages will be automatically deleted after {new_value} days."
-            )
-            notification_string = notification_string.format(
-                user=user_mention, old_value=old_value, new_value=new_value
-            )
+            old_retention_period = f"{old_value} days"
+            new_retention_period = f"{new_value} days"
+            summary_line = f"Messages will be automatically deleted after {new_value} days."
+        notification_string = _(
+            "{user} has changed the [message retention period](/help/message-retention-policy) for this stream:\n"
+            "* **Old retention period**: {old_retention_period}\n"
+            "* **New retention period**: {new_retention_period}\n"
+            "{summary_line}"
```

Comment:
The summary line needs a newline before it in order to appear nicely.  But I also think we should potentially delete it; @alya thoughts?  My concern with saying "Messages will automatically be deleted after {N} days` is that it kinda mis-explains the model, which is that messages will be deleted N days after they are sent, not N days after you change the retention period. Another option would be to extend the string, but I'm kinda feeling like these summary lines are redundant with a very clear statement of what's happening just before them.

Reply:
This is how it looks with a new line before Summary:

![Screenshot from 2022-02-01 21-42-44](https://user-images.githubusercontent.com/451188/152006708-b5f16f2e-5c37-4bc3-9f98-ee1b6fdccdf9.png)

Summary is repeating the information, yes, but the retention policy for a stream is an important notification that I think is okay to be repeated.

Reply:
Yeah, I think it might be worth repeating in this case. How about "Messages in this stream will be automatically deleted 2 days after they are sent." ? That should clear up the ambiguity.

Reply:
Thank you @alya and @timabbott for the input. I have updated the `Message` string at the end of the notification. Let me know if it is okay.

Reply:
Thanks! By the way, do we have a special case for "1 day", so that it doesn't say "1 days"?

Reply:
Not yet, I can pick it up but I think that changeset will pull in translations, because there also we don't make this distinction.